src/plugins/proposal.ts

```diff
@@ -0,0 +1,458 @@
+import BN = require('bn.js')
+import gql from 'graphql-tag'
+import { Observable } from 'rxjs'
+import { IEntityRef, Entity } from '../entity'
+import { IApolloQueryOptions, Address, ICommonQueryOptions } from '../types'
+import { Plugin } from './plugin'
+import { IGenesisProtocolParams, mapGenesisProtocolParams } from '../genesisProtocol'
+import { Arc } from '../arc'
+import { createGraphQlQuery, isAddress, realMathToNumber, hexStringToUint8Array, concat } from '../utils'
+import { IObservable } from '../graphnode'
+import { IVoteQueryOptions, Vote } from '../vote'
+import { Stake, IStakeQueryOptions } from '../stake'
+import { Queue, IQueueState } from '../queue'
+import { IRewardQueryOptions, Reward } from '../reward'
+import { DAO } from '../dao'
+import { ProposalTypeNames, Proposals } from '.'
```

Comment:
We should add a section in the developer docs explaining what needs to be modified when you add a new: Plugin, Proposal Plugin, and Proposal.